### PR TITLE
Update allennlp to latest master, fix tests

### PR DESCRIPTION
**Pull request by matt-gardner**
This fixes all tests in this repo, but it relies on https://github.com/allenai/allennlp/pull/3695.  Once that PR is merged to master, the commit should be updated in this PR, and then this can be merged.  It basically just removes the `num_tokens` from the sorting keys in the bucket iterator specification, which I removed in an earlier PR to allennlp.  There's also one other minor change in `DialogQA`.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # To be changed to allennlp>=1.0 once that's released.
-git+https://github.com/allenai/allennlp.git@8bc5da2d7d0050374c78b52adde7149425e3b52a
+git+https://github.com/allenai/allennlp.git@644db4998487580fa054d28c32ad3dd8622ad6cb
```
Comment (matt-gardner) [inline requirements.txt]: That SHA is going to change again, as we squash merge all PRs to allennlp.

Comment (dirkgr) [reply requirements.txt]: Yes. And it doesn't work like this because it should say `matt-gardner`, not `allenai`. So I guess we'll have to wait.

Comment (dirkgr) [reply requirements.txt]: Actually, it merged just now.

Comment (matt-gardner) [reply requirements.txt]: It's merged now.  I need to finish an ACL review; feel free to update this and merge yourself when you're happy with it.  Not sure if I'll get back to github today.

**Pull request merged**